### PR TITLE
Enable RpaasInstance shutting down using Shutdown flag

### PR DESCRIPTION
**Pull request by gvicentin**

### CODE REVIEW

File: api/v1alpha1/rpaasinstance_types.go [hunk 1]
```diff
@@ -134,6 +134,12 @@ type RpaasInstanceSpec struct {
 	// +optional
 	// +kubebuilder:default:=false
 	Suspend *bool `json:"suspend,omitempty"`
+
+	// Shutdown flag tells whether controller should scale down Nginx instances.
+	// Any assosciated HorizontalPodAutoscaler is remove/created when this flag is toggled.
+	// +optional
+	// +kubebuilder:default:=false
+	Shutdown *bool `json:"shutdown,omitempty"`
```
Comment (wpjunior) [inline api/v1alpha1/rpaasinstance_types.go:142]: @gvicentin I think that just use `bool` will be fine:
when you use `*bool` you have three possible values [null, &false, &true]

when you use `bool` you have just two possible values [false, true]
this is fine and reduce liability of dealing with pointers.

Comment (gvicentin) [reply api/v1alpha1/rpaasinstance_types.go]: Agreed, thanks @wpjunior!

File: controllers/controller_test.go
Comment (morpheu) [inline controllers/controller_test.go:1]: I think it's also a good idea to include tests on Test_reconcileHPA to ensure HPA behavior for both native and KEDA autoscales

File: controllers/controller.go [hunk 2]
```diff
@@ -1186,7 +1190,11 @@ func newNginx(instanceMergedWithFlavors *v1alpha1.RpaasInstance, plan *v1alpha1.
 	}
 
 	replicas := instanceMergedWithFlavors.Spec.Replicas
-	if isAutoscaleEnabled(instanceMergedWithFlavors.Spec.Autoscale) {
+	if shutdown := instanceMergedWithFlavors.Spec.Shutdown; shutdown {
+		*replicas = 0
```
Comment (wpjunior) [inline controllers/controller.go:1194]: in case of repicas is nil, may crash, you could replace to:
```
zero := 0
replicas = &zero
```

Comment (wpjunior) [reply controllers/controller.go:1194]: or use the pointer lib: https://pkg.go.dev/k8s.io/utils@v0.0.0-20230220204549-a5ecb0141aa5/pointer

**Pull request merged**